sky/packages/sky/lib/widgets/dismissable.dart

```diff
@@ -91,38 +83,27 @@ class Dismissable extends AnimatedComponent {
   void _startResizePerformance() {
     assert(_size != null);
     assert(_fadePerformance != null);
+    assert(_fadePerformance.isCompleted);
     assert(_resizePerformance == null);
 
-    // TODO(hansmuller): _fadePerformance is completed; stop shouldn't be needed.
-    _fadePerformance.stop();
-
-    AnimatedValue<double> dismissHeight = new AnimatedValue<double>(_size.height,
-      end: 0.0,
-      curve: ease,
-      interval: new Interval(_kCardDismissResizeDelay, 1.0)
-    );
-    _resizePerformance = new AnimationPerformance()
-      ..variable = dismissHeight
-      ..duration = _kCardDismissResize
-      ..addListener(_handleResizeProgressChanged)
-      ..play();
+    setState(() {
+      _resizePerformance = new AnimationPerformance()
+        ..duration = _kCardDismissResize
+        ..addListener(_handleResizeProgressChanged);
+    });
   }
 
   void _handleResizeProgressChanged() {
-    setState(() {
-      if (_resizePerformance.isCompleted)
-        _maybeCallOnDismissed();
-      else
-        _maybeCallOnResized();
-    });
+    if (_resizePerformance.isCompleted)
+      _maybeCallOnDismissed();
+    else
+      _maybeCallOnResized();
   }
 
   void _handlePointerDown(sky.PointerEvent event) {
-    setState(() {
-      _dragUnderway = true;
-      _dragX = 0.0;
-      _fadePerformance.progress = 0.0;
-    });
+    _dragUnderway = true;
+    _dragX = 0.0;
+    _fadePerformance.progress = 0.0;
```

Comment:
The slide/fade transitions rebuild themselves as the _fadePerformance changes. The rest of the component doesn't need to rebuild.